src/cairoUtilFuncGen/calldata/implicitArrayConversion.ts

```diff
@@ -0,0 +1,503 @@
+          ].join('\n');
+        } else {
+          code = [
+            `    ${this.getOrCreate(targetElmType, sourceElmType)}(${add(
+              'storage_loc',
+              offset,
+            )}, arg[${index}])`,
+          ].join('\n');
+        }
+      }
+      offset = offset + cairoTargetElementType.width;
+      return code;
+    });
+
+    // TODO check implicit order does not matter. //Does not matter.
+    const implicit =
+      '{syscall_ptr : felt*, range_check_ptr, pedersen_ptr : HashBuiltin*, bitwise_ptr : BitwiseBuiltin*}';
+    const code = [
+      `func ${funcName}${implicit}(storage_loc: felt, arg: ${cairoSourceTypeString}):`,
+      `alloc_locals`,
+      ...writeCode,
+      '    return ()',
+      'end',
+    ].join('\n');
+
+    this.generatedFunctions.set(key, { name: funcName, code: code });
+    return { name: funcName, code: code };
+  }
+  //////////////////////////////////////////////////////////
+  //////////////////////////////////////////////////////////
+  //////////////////////////////////////////////////////////
+  private staticToDynamicConversion(
+    key: string,
+    targetType: TypeNode,
+    sourceType: TypeNode,
+  ): CairoFunction {
+    assert(targetType instanceof PointerType && sourceType instanceof PointerType);
+    assert(targetType.to instanceof ArrayType && sourceType.to instanceof ArrayType);
+
+    assert(targetType.to.size === undefined && sourceType.to.size !== undefined);
+
+    const targetElmType = targetType.to.elementT;
+    const sourceElmType = sourceType.to.elementT;
+
+    const funcName = `CD_ST_TO_WS_DY${this.generatedFunctions.size}`;
+    this.generatedFunctions.set(key, { name: funcName, code: '' });
+
+    const cairoTargetElementType = CairoType.fromSol(
+      targetType.to.elementT,
+      this.ast,
+      TypeConversionContext.StorageAllocation,
+    );
+
+    const cairoSourceType = CairoType.fromSol(
+      sourceType,
+      this.ast,
+      TypeConversionContext.CallDataRef,
+    );
+
+    const cairoSourceTypeString = cairoSourceType.toString();
+
+    const sizeSource = narrowBigIntSafe(sourceType.to.size);
+
+    assert(sizeSource !== undefined);
+    // Make sure these are the right way round.
+
+    const dynArrayLengthName = this.dynArrayGen.gen(cairoTargetElementType)[1];
+    const writeCode = mapRange(sizeSource, (index) => {
+      let code;
+      if (targetElmType instanceof IntType) {
+        assert(sourceElmType instanceof IntType);
+        if (targetElmType.nBits === sourceElmType.nBits) {
+          code = [
+            `    let (storage_loc${index}) = ${this.dynArrayIndexAccessGen.getOrCreate(
+              targetElmType,
+            )}(ref, Uint256(${index}, 0))`,
+            `    ${this.storageWriteGen.getOrCreate(
+              targetElmType,
+            )}(storage_loc${index}, arg[${index}])`,
+          ].join('\n');
+        } else if (targetElmType.signed) {
+          // TODO find util to convert index to Uint256
```

Comment:
in `warplib/utils` there is function `uint256` that fits your need here

Reply:
Yeah I was asking Dom about that last night hahaha. I forgot where to find it.